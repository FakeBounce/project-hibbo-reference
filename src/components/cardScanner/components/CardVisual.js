import React from 'react';
import { string } from 'prop-types';

import CardVisualValues from './CardVisualValues';

import * as Styles from '../styles';

import { cards, banks } from '../cardVisualDatas';

const cardNb = Math.floor(Math.random() * cards.length);

const CardVisual = ({
  name,
  cardNumber,
  expiracyMonth,
  expiracyYear,
  cardType,
  crypto,
}) => {
  return (
    <Styles.SWrapper>
      <Styles.SCardVisual source={cards[cardNb]}>
        {cardType !== '' &&
          banks[cardType] && (
            <Styles.SBankIcon
              style={{
                width: banks[cardType].width,
                height: banks[cardType].height,
              }}
              source={banks[cardType].icon}
            />
          )}

        <CardVisualValues
          name={name}
          cardNumber={cardNumber}
          expiracyMonth={expiracyMonth}
          expiracyYear={expiracyYear}
          crypto={crypto}
        />
      </Styles.SCardVisual>
    </Styles.SWrapper>
  );
};

CardVisual.propTypes = {
  cardNumber: string.isRequired,
  expiracyMonth: string.isRequired,
  expiracyYear: string.isRequired,
  name: string.isRequired,
  crypto: string.isRequired,
  cardType: string.isRequired,
};

export default CardVisual;
