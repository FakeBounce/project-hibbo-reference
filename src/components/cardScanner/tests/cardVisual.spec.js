import React from 'react';
import renderer from 'react-test-renderer';

import CardVisual from '../components/CardVisual';

describe('CardVisual', () => {
  let component = renderer.create(
    <CardVisual
      cardNumber=""
      expiracyMonth=""
      expiracyYear=""
      cardType=""
    />,
  ).toJSON();

  it('should render the component', () => {
    expect(component).toMatchSnapshot();
  });

  it('should render the component for a visa card', () => {
    component = renderer.create(
      <CardVisual
        cardNumber=""
        expiracyMonth=""
        expiracyYear=""
        cardType="visa"
      />,
    ).toJSON();

    expect(component).toMatchSnapshot();
  });
});
