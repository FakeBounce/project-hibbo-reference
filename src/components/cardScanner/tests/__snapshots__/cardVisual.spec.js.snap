// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardVisual should render the component 1`] = `
<View
  style={undefined}
>
  <Image
    source={1}
    style={undefined}
  />
  <Image
    source={1}
    style={undefined}
  />
  <Image
    source={1}
    style={undefined}
  />
  <Image
    source={null}
    style={undefined}
  />
  <View
    style={undefined}
  >
    <View
      style={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={undefined}
      >
        
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={undefined}
      >
        
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={undefined}
      >
        
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={undefined}
      >
        
      </Text>
    </View>
    <View
      style={undefined}
    >
      <Image
        source={1}
        style={undefined}
      />
      <View
        style={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={undefined}
        >
          EXPIRE
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={undefined}
        >
          AT END
        </Text>
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={undefined}
      >
        
        /
        
      </Text>
    </View>
  </View>
</View>
`;

exports[`CardVisual should render the component for a visa card 1`] = `
<View
  style={undefined}
>
  <Image
    source={1}
    style={undefined}
  />
  <Image
    source={1}
    style={undefined}
  />
  <Image
    source={1}
    style={undefined}
  />
  <Image
    source={1}
    style={undefined}
  />
  <View
    style={undefined}
  >
    <View
      style={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={undefined}
      >
        
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={undefined}
      >
        
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={undefined}
      >
        
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={undefined}
      >
        
      </Text>
    </View>
    <View
      style={undefined}
    >
      <Image
        source={1}
        style={undefined}
      />
      <View
        style={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={undefined}
        >
          EXPIRE
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={undefined}
        >
          AT END
        </Text>
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={undefined}
      >
        
        /
        
      </Text>
    </View>
  </View>
</View>
`;
